src/lib/tls/tls12/tls_channel_impl_12.cpp

```diff
@@ -371,6 +371,9 @@ size_t Channel_Impl_12::from_peer(std::span<const uint8_t> data) {
    } catch(Decoding_Error&) {
       send_fatal_alert(Alert::DecodeError);
       throw;
+   } catch(Invalid_Argument&) {
+      send_fatal_alert(Alert::IllegalParameter);
```

Comment:
I don't see anything in RFC 8422 or 8446 that indicates what alert to send in this situation. If BoGo has a specific expectation I guess we should match that, but otherwise either of `decrypt_error` or `illegal_parameter` seem reasonable choices.

Reply:
[RFC 8446 Section 6](https://datatracker.ietf.org/doc/html/rfc8446#section-6) is probably authorative here:

> Peers which receive a message which
   cannot be parsed according to the syntax (e.g., have a length
   extending beyond the message boundary or contain an out-of-range
   length) MUST terminate the connection with a "decode_error" alert.
   Peers which receive a message which is syntactically correct but
   semantically invalid (e.g., a DHE share of p - 1, or an invalid enum)
   MUST terminate the connection with an "illegal_parameter" alert.

BoGo switched to "illegal_parameter" recently: https://github.com/google/boringssl/commit/72a60506ded3407454d6ddc1d848c266020c0c82